### ISSUE TITLE
Caching object storage to help a team sharing compute resources

### ISSUE THREAD

**Issue by consideRatio** [pangeo-data/jupyter-earth#31]
This issue represent a work item _to explore_ what kind of object storage caching solutions are out there.

The idea is to help a team using a common JupyterHub for example, to help each other cache requests to a remote object storage within the local cloud or HPC cluster.

**Comment by lheagy** [comment]
A related project to look into here is [pooch](https://github.com/fatiando/pooch). I see @andersy005 has contributed, so he might have some ideas to add!

**Comment by choldgraf** [comment]
+1 to this, I suspect it'll be a common issue for teams that are working from the same cloud dataset and a bunch of 2i2c collaborators would benefit from common patterns here.

Just to concretize a bit - the basic idea is that we'd want to cache *locally* so that the first time a person queried some data it was slow (and consumed bandwidth), but the *next* time a *different* person queried it, it would be "fast" if they queried data that was cached?

**Comment by consideRatio** [comment]
> Just to concretize a bit - the basic idea is that we'd want to cache locally so that the first time a person queried some data it was slow (and consumed bandwidth), but the next time a different person queried it, it would be "fast" if they queried data that was cached?

Yes, where locally should be interpreted as the local _shared_ environment, next to where a jupyterhub is running or simular!

Hmmm perhaps mirroring is a relevant tech to explore in this context also, but we dont want to mirror everything of course, so it would then be a selective mirror, which is almost the definition of cache if the selection is done based on past usage.

**Comment by choldgraf** [comment]
I think that a semi-common use case might be worth highlighting as it's relevant to this:

Say that you are working with data generated from an earth-wide model. It could be that you / your team only cares about a small region within the earth, and you plan to pull from the earth-wide dataset but only work on a small subset. You would benefit from having *super fast* access to a pre-defined subset, even though technically that subset is part of a much larger / slower-to-access dataset.
